New package: python3-loguru-0.6.0

oliver-cfc opened the pull request:
<!-- Uncomment relevant sections and delete options which are not applicable -->

#### Testing the changes
- I tested the changes in this PR: YES

<!--
#### New package
- This new package conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements): YES
-->

<!--
#### Local build testing
- I built this PR locally for my native architecture, (ARCH-LIBC)
-->

classabbyamp commented on srcpkgs/python3-loguru/template:
```diff
@@ -0,0 +1,20 @@
+}
+
+do_check() {
+	python3 setup.py test
```
this is already the default behaviour:

https://github.com/void-linux/void-packages/blob/2859518608b4f682abd4afb6b9f8c2dce01a0523/common/build-style/python3-module.sh#L40-L41

classabbyamp replied:
but in this case you'd want to add `checkdepends="python3-pytest"` because the tests in this package are run with pytest

oliver-cfc replied:
Thanks - I really appreciate your comments and insight. 
Is there an appropriate way to test this before amending the PR?  The original template compiled perfectly in a freshly created xbps-src environment but then failed the build checks.

Does it even need a do_check{} section?

classabbyamp replied:
running xbps-src pkg and xbps-src check would be how I'd test building the template

classabbyamp commented:
the CI fail is probably caused by needing `python3-setuptools` in `hostmakedepends`

there's also a `changelog` that can be added: https://raw.githubusercontent.com/Delgan/loguru/master/CHANGELOG.rst

oliver-cfc commented:
I think this is the good one.  I added python3-colorama to checkdepends (along with python3-pytest) because xbps-src check fails without it but it's not needed for python3-loguru to run. Also added hostmakedepends and checkdepends.  

```
# Template file for 'python3-loguru'
pkgname=python3-loguru
version=0.6.0
revision=1
wrksrc="loguru-${version}"
build_style=python3-module
hostmakedepends="python3-setuptools"
checkdepends="python3-pytest python3-colorama"
short_desc="Library which aims to bring enjoyable logging in Python"
maintainer="oliver <oliver@tuta.io>"
license="MIT"
homepage="https://github.com/Delgan/loguru"
changelog="https://raw.githubusercontent.com/Delgan/loguru/master/CHANGELOG.rst"
distfiles="${PYPI_SITE}/l/loguru/loguru-$version.tar.gz"
checksum=066bd06758d0a513e9836fd9c6b5a75bfb3fd36841f4b996bc60b547a309d41c

post_install() {
        vlicense LICENSE
}
```

Are these the correct next steps (while in my branch)?
1) git add ./srcpkgs/python3-loguru/template
2) git commit --amend
3) git rebase -i HEAD~2
4) git push origin <my branch>

classabbyamp commented:
> Are these the correct next steps (while in my branch)?

```
git add srcpkgs/python3-loguru/template
git commit --amend
git push --force
```
should be enough

oliver-cfc commented:
I assume I'm hitting this: https://github.com/Delgan/loguru/issues/658
Do you want this PR closed until a new release of loguru?

classabbyamp commented:
i think it would be better to just disable that test (see pytest docs and use `make_check_args`) if it's a known and fixed issue in this version
